### ISSUE TITLE
conflicting implementation false positive for fix point over GAT

### ISSUE THREAD

**Issue by inanna-malick** [rust-lang/rust#115918]
I tried this code (on stable 1.72 and nightly 1.74):

```rust
pub trait HasFrame {
    type Frame<X>;
}

pub struct Fix<F: HasFrame>(pub Box<F::Frame<Fix<F>>>);

// conflicts with impl<T> From<T> for T;
impl<F> From<F::Frame<Fix<F>>> for Fix<F> where F: HasFrame {
}
```

I expected it to fail to compile because `fn from` isn't implemented on `From`.

Instead, it failed to compile with error:
```
Compiling playground v0.0.1 (/playground)
error[[E0119]](https://doc.rust-lang.org/stable/error_codes/E0119.html): conflicting implementations of trait `From<Fix<_>>` for type `Fix<_>`
  --> src/lib.rs:10:1
   |
10 | impl<F> From<F::Frame<Fix<F>>> for Fix<F> where F: HasFrame {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T> From<T> for T;

For more information about this error, try `rustc --explain E0119`.
error: could not compile `playground` (lib) due to previous error

stable (Stable version: 1.72.0): https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2c8aa9d2e05aee713e7fb6ebb6c640d9

nightly (Nightly version: 1.74.0-nightly 2023-09-16 7d9bce327332fef22cf6)  https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=1fc6227e895326caab05bee1627019aa

**Comment by inanna-malick** [comment]
nvm - this is because there's no way to tell that `F::Frame<Fix<F>>` wont end up as the same type as `Fix<F>`
